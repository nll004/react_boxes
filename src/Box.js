import React from "react";

function Box({backgroundColor, width, height}) {
    return (
        <div style={{backgroundColor: backgroundColor, width: width, height: height}}></div>
    )
}

export default Box;
