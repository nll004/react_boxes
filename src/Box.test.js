import React from "react";
import { render } from "@testing-library/react";
import Box from "./Box";

const testBoxComponent = '<Box color="red" width="200px" height="300px" data-testid="1" />'

it('renders Box component without crashing', ()=> {
    render(<Box />);
    render(testBoxComponent);
})

it('matches snapshot', ()=> {
    const { asFragment } = render(testBoxComponent);
    expect(asFragment()).toMatchSnapshot();
})
