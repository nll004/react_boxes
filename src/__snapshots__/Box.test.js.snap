// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  &lt;Box color="red" width="200px" height="300px" data-testid="1" /&gt;
</DocumentFragment>
`;
